### ISSUE TITLE
Support inline button actions in popover menu

### ISSUE THREAD

**Issue by skjnldsv** [nextcloud/nextcloud-vue#417]
@jancborchardt what do you think?
@nextcloud/designers

**Comment by jancborchardt** [comment]
What do you mean by that? Do you have a further explanation or a mockup? :)

**Comment by skjnldsv** [comment]
like 3 actions in a single popovermenu row :)
instead of 3 rows

**Comment by jancborchardt** [comment]
Right, as we discussed for the folder permissions. :)

So as always, I don’t think we should start by standardizing before we know it’s actually useful in many cases. Especially because right now we only know we’ll use it for one case. We should just go ahead and build it in that one place in futureproof markup, and then once we use it in 2–3 places we can standardize based on what we learnt.

What do you think?

**Comment by skjnldsv** [comment]
Yes, it was required somewhere else on spreed iirc, for the notification select.
Basically this would be used for radio behaviour. One set of options where only one is selectable.

**Comment by jancborchardt** [comment]
@skjnldsv right, that’s a similar case. Actually looking at the Talk menu too makes me question whether this 3 on the same row is the right choice, as we won’t have enough space for the text then.

So maybe the way to go is:
- Keep the Talk menu like it is, cause it’s quite straightforward.
- For the share menu, introduce a divider border below the radio choice just like we have in the Talk app to separate the main choice from the rest. Also we could think about icons for the different choices. And currently the blue bar on the left is missing, which we have for the chosen option in the Talk app.
- Also, if we ever need more space for explanatory text on any of these options, the list is much better than 3 on the same row. Just like we do it in the personal settings for federation visibility.

![talk app menu](https://user-images.githubusercontent.com/925062/58555481-e0dec580-8219-11e9-9e48-7c91d5667f76.png) ![share folder menu](https://user-images.githubusercontent.com/925062/58555482-e1775c00-8219-11e9-8982-e18d27ccc7ab.png)

**Comment by skjnldsv** [comment]
> whether this 3 on the same row is the right choice, as we won’t have enough space for the text then.

What about three icons in a row and display the selected option text under it?
(shitty mockup)
![58555481-e0dec580-8219-11e9-9e48-7c91d5667f76](https://user-images.githubusercontent.com/14975046/58556571-9f9be500-821c-11e9-8ba0-660c294f4262.png)

**Comment by jancborchardt** [comment]
This is basically similar to what I imagined, why I then wrote my comment above. ;) Several issues with it:
- You only see the text of the currently selected item, so it’s basically [mystery meat navigation](https://en.m.wikipedia.org/wiki/Mystery_meat_navigation)
- It doesn’t really save a lot of space, only 1 row in this case.
- Suddenly the layout goes from vertical to horizontal
- The method of picking is completely different than anywhere else.

That’s why I came to the conclusion I wrote above at https://github.com/nextcloud/nextcloud-vue/issues/417#issuecomment-496906500 :)
